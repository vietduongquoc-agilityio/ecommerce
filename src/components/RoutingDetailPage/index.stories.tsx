import { Meta, StoryObj } from "@storybook/react";
import RoutingDetailPage from "./";

const meta: Meta<typeof RoutingDetailPage> = {
  title: "Components/RoutingDetailPage",
  component: RoutingDetailPage,
  tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof RoutingDetailPage>;

export const Default: Story = {};
