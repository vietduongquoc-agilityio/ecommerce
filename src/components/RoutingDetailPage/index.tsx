import Image from "next/image";
import { fonts, colors } from "@/themes";
import arrowDown from "@/assets/Images/Arrow-down.png";
import line from "@/assets/Images/Line.png";
import { Product } from "@/Interface/product";

interface RoutingDetailPageProps {
  product: Product;
}

const RoutingDetailPage = ({ product }: RoutingDetailPageProps) => {
  return (
    <section
      style={{
        width: "100%",
        height: "auto",
        background: colors.primary.dark,
        padding: "30px 100px",
        display: "flex",
        justifyContent: "space-between",
        alignItems: "center",
        marginTop: "10px",
      }}
    >
      <div
        style={{
          display: "flex",
          justifyContent: "space-between",
          width: "350px",
          height: "37px",
          alignItems: "center",
          fontSize: fonts.size.xs,
          fontWeight: fonts.weight.regular,
        }}
      >
        <p
          style={{
            color: colors.secondary.splight,
          }}
        >
          Home
        </p>
        <Image src={arrowDown} alt={"arrowDown"} />
        <p
          style={{
            color: colors.secondary.splight,
          }}
        >
          Shop
        </p>
        <Image src={arrowDown} alt={"arrowDown"} />
        <Image src={line} alt={"line"} />
        <p
          style={{
            color: colors.black,
          }}
        >
          {product.productName}
        </p>
      </div>
    </section>
  );
};

export default RoutingDetailPage;
