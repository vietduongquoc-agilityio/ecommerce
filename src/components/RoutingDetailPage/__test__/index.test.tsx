import { render, screen } from "@testing-library/react";
import RoutingDetailPage from "../";

describe("RoutingDetailPage", () => {
  it("should render correctly and match snapshot", () => {
    const { asFragment } = render(<RoutingDetailPage />);
    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByText("Shop")).toBeInTheDocument();
    expect(screen.getByText("Asgaard sofa")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
