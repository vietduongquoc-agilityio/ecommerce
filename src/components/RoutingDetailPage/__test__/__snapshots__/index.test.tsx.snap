// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutingDetailPage should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <section
    style="width: 100%; height: auto; background: rgb(249, 241, 231); padding: 30px 100px; display: flex; justify-content: space-between; align-items: center; margin-top: 10px;"
  >
    <div
      style="display: flex; justify-content: space-between; width: 350px; height: 37px; align-items: center; font-size: 16px; font-weight: 400;"
    >
      <p
        style="color: rgb(159, 159, 159);"
      >
        Home
      </p>
      <img
        alt="arrowDown"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <p
        style="color: rgb(159, 159, 159);"
      >
        Shop
      </p>
      <img
        alt="arrowDown"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <img
        alt="line"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <p
        style="color: rgb(0, 0, 0);"
      >
        Asgaard sofa
      </p>
    </div>
  </section>
</DocumentFragment>
`;
